cmd/nerdctl/run.go

```diff
@@ -1056,6 +1057,10 @@ func withInternalLabels(ns, name, hostname, containerStateDir string, extraHosts
 		m[labels.Mounts] = string(mountPointsJSON)
 	}
 
+	if macAddress != "" {
```

Comment:
For my own understanding why do we need labels?

Reply:
Required when oci hooks

Reply:
Because the way we set MAC addresses is by tagging it on the label, then actually configuring it when OCI hooks are executed.